sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -225,6 +225,11 @@ case class AlterTableSetPropertiesCommand(
     val catalog = sparkSession.sessionState.catalog
     val table = catalog.getTableMetadata(tableName)
     DDLUtils.verifyAlterTableType(catalog, table, isView)
+    // Not allowed to switch the table type.
+    if (properties.contains("EXTERNAL")) {
```

Comment:
In the initial fix, it is only for Hive. : )

Moving it to `verifyTableProperties` sounds good to me. It also covers another case. Users are unable to set `EXTERNAL` in `CREATE TABLE`. Let me do it. Thanks!